docs/user/advanced.rst

```diff
@@ -64,26 +64,21 @@ Some applications render templates based on data sources (e.g. CSVs or
 JSON files).
 
 To get data to templates you can set up a mapping between filenames
-and functions which generate dictionaries containing the data:
+and dictionaries or functions which generate dictionaries containing 
+the data:
 
 .. code-block:: python
 
     from staticjinja import make_site
 
-    def get_knights():
-        """Generate knights of the round table."""
-        knights = [
-            'sir arthur',
-            'sir lancelot',
-            'sir galahad',
-        ]
-        return {'knights': knights}
-
     if __name__ == "__main__":
         site = make_site(contexts=[
-            ('index.html', get_knights),
+            (
+             'index.html', 
+             { 'knights': ['sir arthur', 'sir lancelot', 'sir galahad'] }
```

Comment:
I had this concern too. But when I first read the documentation I really thought it was an insanely convoluted way of doing things in the knights example. I think the current workflow is complicated both for naive users (here I mean someone who barely knows python and wants to generate a very simple site where the only use of jinja is to avoid repeating boilerplate code and hardcoding a few urls), for veteran jinja users (where variables are passed as named arguments rather than a dict or function) and for veteran django users who still want simplicity sometimes (see e.g. https://docs.djangoproject.com/en/1.9/intro/overview/#write-your-views to see how template variables are passed as a dictionary there).

 And I think there are three levels of complexity anyway. At level one you don't have a custom build script at all. At level two you simply want to set the output path and pass a few urls and content to templates. At level three you write custom rules, need the template to generate the context, write filters etc. I see the dictionary thing living at level 2 whereas the function with or without argument is already for different users at level 3.

That being said I perfectly understand that keeping staticjinja simple is crucial and, as I stated right away, I think the ability to merge contexts from multiple regex is much more important. It's really up to you. 

If you choose to remove this new way then I think it would be helpful if the documentation does the knights thing with a lambda, introduces an example showing a case where you need a more complicated function but without using the template argument and then has an example where the argument is used. By the way, the example I added is the only case I used this possibility in my code but I'd be curious to read any other example.